### ISSUE TITLE
read token from cookie

### ISSUE THREAD

**Issue by mckinlde** [PostgREST/postgrest#3050]
### Environment

* PostgreSQL version: 15 (stable)
* PostgREST version: 11.2 (stable)
* Operating system: AMI Linux 2023 on prod, macOS Monterey 12.7 in dev.

### Description of issue

[After](https://github.com/PostgREST/postgrest/issues/940) [literature](https://github.com/PostgREST/postgrest-docs/issues/240) [review](https://github.com/PostgREST/postgrest/issues/773), it seems that to implement cookie-based auth requires a reverse proxy server for the sole purpose of handling requests.  Specifically, a reverse proxy just for reading `cookies` and remaking requests with an `Authorization bearer=token` header.  This is despite PostgREST already being capable of issuing JWTs to a `cookie` directly as in this [snippet](https://github.com/PostgREST/postgrest/issues/773#issuecomment-451037544), and suggested in [JWT Security](https://postgrest.org/en/stable/references/auth.html#jwt-security).

As a result, novice end users fresh off [SQL User Management](https://postgrest.org/en/stable/how-tos/sql-user-management.html) and [Tutorial 1](https://postgrest.org/en/v11.0/tutorials/tut1.html) are unable to implement a persistent auth system without opening another port, and taking on configuration of reverse proxies.  Currently both those tutorials and the resulting [client-auth](https://postgrest.org/en/stable/references/auth.html#client-auth) login system can be implemented such that PostgREST is serving all requests directly.  

It is currently supported to serve the JWT to an HttpOnly `cookie` with a minor addition to the login(text, text) function outlined in SQL User Management using a PostgREST-native [response header setting](https://postgrest.org/en/v9.0/api.html#setting-response-headers) implemented in [#986](https://github.com/PostgREST/postgrest/pull/986), but once that is done there is not a way to configure [client-auth](https://postgrest.org/en/stable/references/auth.html#client-auth) to check in a location other than the `Authorization` header for the JWT.  With a configuration setting to read JWTs from a `cookie`, these users could secure their web sessions with only the PostgREST server.

This issue has been previously discussed in [2016](https://github.com/PostgREST/postgrest/issues/773) and [2017](https://github.com/PostgREST/postgrest/issues/940), specifically for the purpose of cookie-based auth.

### Expected behavior

The [Custom Validation](https://postgrest.org/en/stable/references/auth.html#custom-validation) section, or possibly a tutorial 2, would describe configuring the SQL User Management system to issue and check JWTs as `cookies`.

 ### vs actual behavior

Issuing JWTs to cookies can be implemented in SQL User Management's login() function per the [snippet](https://github.com/PostgREST/postgrest/issues/773#issuecomment-451037544), but a separate reverse proxy server must be configured and maintained to transcribe tokens from `cookies` to `Authorization` headers on all incoming requests.

### (Steps to reproduce: Include a minimal SQL definition plus how you make the request to PostgREST and the response body)

Minimal SQL definition is more or less the content of [SQL User Management](https://postgrest.org/en/v11.0/how-tos/sql-user-management.html) and the [snippet](https://github.com/PostgREST/postgrest/issues/773#issuecomment-451037544).  The issue is not that PostgrREST responds with a malformed body, but that it does not (to my knowledge) support checking for a `cookie`, instead of an `Authorization` header.

I try to include all relevant documentation and previous discussions in this post, but the result is wordy.  Editorial suggestions welcome!

**Comment by mckinlde** [comment]
This issue is made irrelevant for my use case with a better understanding of refresh and access tokens, per https://stackoverflow.com/questions/27726066/jwt-refresh-token-flow

I'd still like to write a Tutorial 2: Automatic token refresh / graceful token expiry, but no further development in PostgREST is needed to do that, the existing [response header setting](https://postgrest.org/en/v9.0/api.html#setting-response-headers) implemented in https://github.com/PostgREST/postgrest/pull/986 is adequate!

**Comment by mckinlde** [comment]
> This is basically a duplicate of #3033

Yes
